libraries/standard/json/cbmc/proofs/JSON_Search/glue.c

```diff
@@ -0,0 +1,53 @@
+#include "glue.h"
+
+bool_ skipAnyScalar( const char * buf,
+                     size_t * start,
+                     size_t max )
+{
+    /* min argument is 1 for a single character scalar like 0,
+     * or 2 for an empty double-quoted string, i.e., "". */
+    size_t min = 1;
+
+    if( ( *start < max ) && ( buf[ *start ] == '"' ) )
```

Comment:
Same comment about using __CPROVER_assert to check that start and buf are not NULL

Reply:
There's already a __CPROVER_r_ok on the buffer just after the malloc calls in the harness.  The real skipAnyScalar has a (regular) assert within, and it's proof runs cleanly, by which I reason I do not need one here since I only need to duplicate the index manipulations for the purpose of proving JSON_Search().